### PR TITLE
[flang] Sync TypeCode.h after upstreaming review

### PR DESCRIPTION
**Pull request by clementval**

### CODE REVIEW

File: flang/include/flang/Optimizer/Support/TypeCode.h [hunk 1]
```diff
@@ -34,22 +34,21 @@ inline int characterBitsToTypeCode(unsigned bits) {
...
   // clang-format off
-  switch (bits) {
-  case 16:  return CFI_type_half_float_Complex; // CFI_type_bfloat_Complex ?
+  switch (bitwidth) {
   case 32:  return CFI_type_float_Complex;
   case 64:  return CFI_type_double_Complex;
-  case 80:  return CFI_type_extended_double_Complex;
-  case 128: return CFI_type_float128_Complex;
+  case 80:
+  case 128: return CFI_type_long_double_Complex;
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/Support/TypeCode.h:43]: This appears to be backwards. Didn't @vdonaldson just add the new CFI codes?

Comment (clementval) [reply flang/include/flang/Optimizer/Support/TypeCode.h:43]: Was not there when this was upstreamed I guess. This sync is from smth upstreamed back in November or December.

Comment (clementval) [reply flang/include/flang/Optimizer/Support/TypeCode.h:43]: Should we use `CFI_type_extended_double_Complex` for `case 80:`?

Comment (clementval) [reply flang/include/flang/Optimizer/Support/TypeCode.h:43]: Ok it's from e43b2e4f480b507a1d3eb9a5f3b294c0599eed73. It was not there before. I'll update this PR and then upstream the change back to main.

Comment (clementval) [reply flang/include/flang/Optimizer/Support/TypeCode.h]: @schweitzpgi I just realized that I did a really bad sync at first. I didn't noticed the changes and just updated it so it should be just a name change right now. It was asked by the reviewer...

Comment (vdonaldson) [reply flang/include/flang/Optimizer/Support/TypeCode.h]: Looks ok to me

**Review by schweitzpgi — APPROVED** [review]
Looks ok now.

The number of bits and a bit width are the same by definition, no? ;)

**Comment by clementval** [comment]
> Looks ok now.
> 
> 
> 
> The number of bits and a bit width are the same by definition, no? ;)

Yeah! At some point arguing takes more time than making the change ;)

**Pull request merged**